forage-android/src/main/java/com/joinforage/forage/android/collect/BasisTheoryPinSubmitter.kt

```diff
@@ -0,0 +1,139 @@
+package com.joinforage.forage.android.collect
+
+import android.content.Context
+import com.basistheory.android.service.BasisTheoryElements
+import com.basistheory.android.service.ProxyRequest
+import com.joinforage.forage.android.VaultType
+import com.joinforage.forage.android.core.StopgapGlobalState
+import com.joinforage.forage.android.model.EncryptionKeys
+import com.joinforage.forage.android.model.PaymentMethod
+import com.joinforage.forage.android.network.ForageConstants
+import com.joinforage.forage.android.network.model.ForageApiError
+import com.joinforage.forage.android.network.model.ForageApiResponse
+import com.joinforage.forage.android.network.model.ForageError
+import com.joinforage.forage.android.network.model.UnknownErrorApiResponse
+import com.joinforage.forage.android.ui.ForagePINEditText
+import org.json.JSONException
+
+internal typealias BasisTheoryResponse = Result<Any?>
+
+internal class BasisTheoryPinSubmitter(
+    context: Context,
+    foragePinEditText: ForagePINEditText
+) : AbstractVaultSubmitter<BasisTheoryResponse>(
+    context = context,
+    foragePinEditText = foragePinEditText,
+    vaultType = VaultType.BT_VAULT_TYPE
+) {
+    override fun parseEncryptionKey(encryptionKeys: EncryptionKeys): String {
+        return encryptionKeys.btAlias
+    }
+
+    // Basis Theory requires a few extra headers beyond the
+    // common headers to make proxy requests
+    override fun buildProxyRequest(
+        encryptionKey: String,
+        idempotencyKey: String,
+        merchantId: String,
+        vaultToken: String
+    ) = super
+        .buildProxyRequest(
+            encryptionKey = encryptionKey,
+            idempotencyKey = idempotencyKey,
+            merchantId = merchantId,
+            vaultToken = vaultToken
+        )
+        .setHeader(ForageConstants.Headers.BT_PROXY_KEY, PROXY_ID)
+        .setHeader(ForageConstants.Headers.CONTENT_TYPE, "application/json")
+
+    override suspend fun submitProxyRequest(vaultProxyRequest: VaultProxyRequest): ForageApiResponse<String> {
+        val bt = buildBt()
+
+        val proxyRequest: ProxyRequest = ProxyRequest().apply {
+            headers = vaultProxyRequest.headers
+            body = ProxyRequestObject(
+                pin = foragePinEditText.getTextElement(),
+                card_number_token = vaultProxyRequest.vaultToken
+            )
+            path = vaultProxyRequest.path
+        }
+
+        val vaultResponse = runCatching {
+            bt.proxy.post(proxyRequest)
+        }
+
+        return vaultToForageResponse(vaultResponse)
+    }
+
+    override fun parseVaultError(vaultResponse: BasisTheoryResponse): String {
+        return vaultResponse.exceptionOrNull().toString()
+    }
+
+    override fun toForageSuccessOrNull(vaultResponse: BasisTheoryResponse): ForageApiResponse.Success<String>? {
```

Comment:
I think this function should be identical to the `toForageErrorOrNull` function. We want to try to parse the response as a successful Forage response and return `null` if we don't get it.

Reply:
6c72e226a80e5eeb580d56240707c80ec835c192